node/src/command.rs

```diff
@@ -122,15 +200,43 @@ pub fn run() -> sc_cli::Result<()> {
             )?;
             let state_version = Cli::native_runtime_version(chain_spec).state_version();
 
-            let block: zeitgeist_runtime::Block =
-                cumulus_client_service::genesis::generate_genesis_block(chain_spec, state_version)?;
-            let raw_header = block.header().encode();
-            let buf = if params.raw {
-                raw_header
-            } else {
-                format!("0x{:?}", HexDisplay::from(&block.header().encode())).into_bytes()
+            let buf = match chain_spec {
+                #[cfg(feature = "with-zeitgeist-runtime")]
+                spec if spec.is_zeitgeist() => {
+                    let block: zeitgeist_runtime::Block =
+                        cumulus_client_service::genesis::generate_genesis_block(
+                            chain_spec,
+                            state_version,
+                        )?;
+                    let raw_header = block.header().encode();
+
+                    if params.raw {
+                        raw_header
+                    } else {
+                        format!("0x{:?}", HexDisplay::from(&block.header().encode())).into_bytes()
+                    }
+                }
+                #[cfg(feature = "with-battery-station-runtime")]
+                _ => {
+                    let block: battery_station_runtime::Block =
+                        cumulus_client_service::genesis::generate_genesis_block(
+                            chain_spec,
+                            state_version,
+                        )?;
+                    let raw_header = block.header().encode();
+
+                    if params.raw {
+                        raw_header
+                    } else {
+                        format!("0x{:?}", HexDisplay::from(&block.header().encode())).into_bytes()
+                    }
+                }
+                #[cfg(not(feature = "with-battery-station-runtime"))]
+                _ => panic!("Invalid chain spec"),
             };
 
+            // let raw_header = block.header().encode();
```

Comment:
remove if not required.

Reply:
Mitigated in [ab798a5](https://github.com/zeitgeistpm/zeitgeist/pull/721/commits/ab798a5059ca11de5b0dd515e4c1d1ba0e9dca7c)